### PR TITLE
Titan DB support.

### PR DESCRIPTION
**Pull request by DarioAmiri**

### CODE REVIEW

**Comment by DarioAmiri** [comment]
NOTE: This is for review only DO NOT MERGE.

File: service/src/main/java/com/ge/predix/acs/privilege/management/dao/GraphGenericRepository.java [hunk 1]
```diff
@@ -0,0 +1,210 @@
...
+    }
+
+    @Override
+    public void deleteInBatch(final Iterable<E> entities) {
+        delete(entities);
+    }
+
+    @Override
+    public List<E> findAll() {
+        return this.graph.traversal().V().has(getEntityIdKey()).toList().stream().map(item -> vertexToEntity(item))
```
Comment (sanjeevchopra) [inline service/src/main/java/com/ge/predix/acs/privilege/management/dao/GraphGenericRepository.java]: welcome java8!

File: service/src/test/java/com/ge/predix/acs/privilege/management/dao/GraphResourceRepositoryTest.java [hunk 2]
```diff
@@ -0,0 +1,223 @@
...
+    public void testGetOrCreateZoneVertex() {
+        assertEquals(IteratorUtils.count(this.graph.vertices()), 0);
+        this.resourceRepository.getOrCreateZoneVertex(TEST_ZONE_1);
+        assertEquals(IteratorUtils.count(this.graph.vertices()), 1);
+        this.resourceRepository.getOrCreateZoneVertex(TEST_ZONE_1);
+        assertEquals(IteratorUtils.count(this.graph.vertices()), 1);
+    }
+
+    @Test
+    public void testGetByZoneAndResourceIdentifier() {
```
Comment (sanjeevchopra) [inline service/src/test/java/com/ge/predix/acs/privilege/management/dao/GraphResourceRepositoryTest.java:137]: Maybe add a  test  with a graph depth more than 2 ?    like:  
zone1 > R1 > R2
           >R4
zone2 > R3

**Comment by sanjeevchopra** [comment]
:+1: 

Great work @amiri-ge  !

File: service/src/test/java/com/ge/predix/acs/privilege/management/dao/GraphResourceRepositoryTest.java (hunk 2 above)
Comment (DarioAmiri) [inline service/src/test/java/com/ge/predix/acs/privilege/management/dao/GraphResourceRepositoryTest.java:137]: Yes, those are the tests we'll have to add eventually once we begin to add capabilities for hierarchical attributes.

File: service/src/main/java/com/ge/predix/acs/privilege/management/dao/GraphGenericRepository.java (hunk 1 above)
Comment (DarioAmiri) [inline service/src/main/java/com/ge/predix/acs/privilege/management/dao/GraphGenericRepository.java]: I know, isn't it cool? Still trying to get the hang of it.

**Comment by DarioAmiri** [comment]
@sanjeevchopra 
Thanks :)

**Comment by bondugula** [comment]
💯

**Comment by sanjeevchopra** [comment]
Jenkins PR build - http://sjc1jenkins08.crd.ge.com:8080/job/Predix-Security/job/acs-build-n-test-pullrequest/582/

**Pull request merged**